cruise.umple/src/GrammarParsing_Code.ump

```diff
@@ -379,9 +527,38 @@ class RuleBasedParser
       }
     }
     analyzer = new GrammarAnalyzer();
+    analyzer.setLinkedFileHandler(getLinkedFileHandler());
+    analyzer.setAnalyzerGenerator(getAnalyzerGenerator());
+    for( Map.Entry<String,ParserAction> entry : actionedTokens.entrySet() ){
+      analyzer.addParserAction(entry.getKey(), entry.getValue());
+    }
     analyzer.analyze(rootToken);
     analyzer.setupTerminals();
   }
+  
+  // retroactively change the linked file handler if an analyzer is already constructed
+  void setLinkedFileHandler(LinkedFileHandler aLinkedFileHandler){
+    linkedFileHandler = aLinkedFileHandler;
+    if( analyzer != null ){
+      analyzer.setLinkedFileHandler(aLinkedFileHandler);
+    }
+  }
+  
+  LinkedFileHandler getLinkedFileHandler(){
```

Comment:
> Due to the choiceRules fields being static, the handlers have to be static in order to work with the initialization around Umple. I added these as member methods for consistency with the current getAnalyzer() method.
I trust your judgement!

> Short of doing a whole rewrite of the parser, which would require a massive restructure to the tests and other parts of Umple, there was no way around dealing with the static variables.
I guess you know your next project. :wink: But in reality, that make sense, I understand.